### PR TITLE
[SYCL][ESIMD] reduce restriction on some esimd API based upon user feedback

### PR DESCRIPTION
**Pull request by cmc-rep**
Signed-off-by: Gang Y Chen <gang.y.chen@intel.com>

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
@cmc-rep, could you please add a short description with an explanation what was done and how?

**Comment by cmc-rep** [comment]
> @cmc-rep, could you please add a short description with an explanation what was done and how?

More description added in the commit comment

**Review by dm-vodopyanov — COMMENTED** [review]
@cmc-rep it's better to put the description to the PR's description.
In common, code LGTM, but not approving yet. I would like @kbobrovs to review as well.

**Review by kbobrovs — COMMENTED** [review]
@cmc-rep, can a test be added for
 - n = 1, n == 2, n == 4 added for gather scatter (newly enabled variants)
 - block size 8 and 16 for slm_block_load/store?

**Comment by cmc-rep** [comment]
> @cmc-rep, can a test be added for
> 
> * n = 1, n == 2, n == 4 added for gather scatter (newly enabled variants)
> * block size 8 and 16 for slm_block_load/store?

Tests added

File: sycl/test/basic_tests/esimd/gather_scatter.cpp [hunk 1]
```diff
@@ -9,7 +9,22 @@
 using namespace sycl::INTEL::gpu;
 using namespace cl::sycl;
 
-void kernel(accessor<int, 1, access::mode::read_write, access::target::global_buffer> &buf) __attribute__((sycl_device)) {
+void kernel0(accessor<int, 1, access::mode::read_write,
```
Comment (kbobrovs) [inline sycl/test/basic_tests/esimd/gather_scatter.cpp:12]:
```suggestion
// TODO: add executable tests for gather/scatter with verification
void kernel0(accessor<int, 1, access::mode::read_write,
```

File: sycl/test/basic_tests/esimd/slm_block.cpp [hunk 2]
```diff
@@ -18,3 +18,13 @@ void kernel() __attribute__((sycl_device)) {
 
   slm_block_store<int, 32>(0, v0);
 }
+
+void kernel2() __attribute__((sycl_device)) {
```
Comment (kbobrovs) [inline sycl/test/basic_tests/esimd/slm_block.cpp:22]:
```suggestion
// TODO: add executable tests for slm load/stores with verification
void kernel2() __attribute__((sycl_device)) {
```

**Pull request closed without merging**